pkg/storage/parallel.go

```diff
@@ -4,25 +4,31 @@ import (
 	"context"
 	"errors"
 
-	"github.com/bacalhau-project/bacalhau/pkg/model"
-	"github.com/bacalhau-project/bacalhau/pkg/util/generic"
 	"github.com/rs/zerolog/log"
 	"go.ptx.dk/multierrgroup"
+
+	"github.com/bacalhau-project/bacalhau/pkg/model"
 )
 
+type PreparedStorage struct {
+	Spec   model.StorageSpec
+	Volume StorageVolume
+}
+
 // ParallelPrepareStorage downloads all of the data necessary for the passed
 // storage specs in parallel, and returns a map of specs to their download
 // volume counterparts.
 func ParallelPrepareStorage(
 	ctx context.Context,
 	provider StorageProvider,
-	specs []model.StorageSpec,
-) (map[*model.StorageSpec]StorageVolume, error) {
-	volumes := generic.SyncMap[*model.StorageSpec, StorageVolume]{}
+	specs ...model.StorageSpec,
+) ([]PreparedStorage, error) {
 	waitgroup := multierrgroup.Group{}
 
-	for _, inputStorageSpec := range specs {
+	out := make([]PreparedStorage, len(specs))
+	for i, inputStorageSpec := range specs {
 		spec := inputStorageSpec // https://golang.org/doc/faq#closures_and_goroutines
+		i := i
```

Comment:
(nit) Instead of shadowing the same name (I first read it as 1 and got confused), should we give it a different name and capture that in the function instead?